docs/dev/kubernetes/kubernetes_basic_workflow.md

```diff
@@ -0,0 +1,29 @@
+# Kubernetes Basic Transformation Workflow
+
+This document outlines the workflow for the deployment of a simple application on the Kubernetes platform in order to help design the basic Kubernetes plugin for TOSCAna.
+
+#### Basic Information about Kubernetes
+- Packs applications in isolated containers instead of VMs (one application per container)
+- Deploys containers based on operating-system-level virtualization rather than hardware virtualization
+- Kubernetes mainly uses Docker for containers
+
+## Automated Workflow
+Describes the planned automated deployment of the simple application based on our [transformation workflow](https://github.com/StuPro-TOSCAna/TOSCAna/blob/transformation-flow/docs/dev/architecture/workflow.md).
+
+### Transformation
+- Data model of the simple app is provided by core
+- Application is already containerized through Docker
+  - **Note**: For the full version of the plugin an additional step is needed in order to containerize applications. This will probably be fairly complex, since the plugin must decide which parts of the application will be put in which container.
+- Plugin creates Kubernetes Deployment configuration (YAML) based on data model
+- **Output**: target artifacts (Deployment configuration, container)
```

Comment:
The problem with kubernetes is, that it needs a image path on a registry in order to deploy the image on kubernetes. This point has to be adressed somewhere

Reply:
I'm not sure if I understand this properly. 

We have a container image and a Kubernetes cluster. In order to deploy the image on this cluster, we must create a Deployment configuration (YAML) file which specifies the location of our container image and other deployment related information. 
So is this location the one which is the path to the container image on the registry? 

Also, this registry is hosted seperately from the Kubernetes cluster and our TOSCAna application, right?

Reply:
Because docker does not feature a format to transport images as files there are only two ways to "move" a docker image to a different machine. The first one is taking the "source" of the docker image and building it whereever we need it.
The other option is to push the docker Image onto a registy and the machine on wich you want to access the image just pulls from the registry. This is the way to go because kubernetes only supports pulling from registires.
For the transformation we have to differenatiate between two different cases:
**First**:
the csar describes a path in a registry. In this case we dont have to do anything except for linking the image in the Kubernetes deployment. (Ignoring authentication for now) because kubernetes will deploy it from this image path.
**Second**:
the csar only features the "source" of the Image. Here we have to build the image (using `docker build` on a appropriate host) and then push it to a registry that has to be supplied by the user. After pushing we got a "URL" that we can use in combination with kubernetes.

To your last question:
Usualy the registry is hosted completely seperately,